src/main/java/com/urswolfer/intellij/plugin/gerrit/push/GerritPushExtensionPanel.java

```diff
@@ -113,7 +113,7 @@ public void run() {
     private Optional<String> getGitReviewBranchName() {
         Optional<String> branchName = Optional.absent();
 
-        DataContext dataContext = DataManager.getInstance().getDataContextFromFocus().getResult();
+        DataContext dataContext = DataManager.getInstance().getDataContext(this);
```

Comment:
Looks good. Have you tested this code with the following situation:
1) push dialog invoked by commit dialog?
2) push dialog invoked by VCS menu?

Have you tried it with the most recent version of IntelliJ?

Reply:
Tried all situations on both default branch (ibtellij14) and latest branch (intellij2016.2). Both works.